base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
+/// Add a transaction output to a transaction struct
 #[no_mangle]
-pub unsafe extern "C" fn add_pending_transaction_outputs(
-    wallet: *mut WalletFfi,
-    pending_tx: *mut PendingTransactionOutputs,
+pub unsafe extern "C" fn transactionoutputs_add_transaction_output(
+    outputs: *mut TransactionOutputs,
+    transaction: *mut TransactionOutput,
 ) -> bool
 {
-    // append this data to the wallet
+    if outputs.is_null() {
+        return false;
+    }
+
+    if transaction.is_null() {
+        return false;
+    }
+
+    (*outputs).0.push((*transaction).clone());
+    return true;
 }
 
-/// Initialize a Transaction struct to be populated
+/// Add a transaction kernel to a transaction struct
 #[no_mangle]
-pub unsafe extern "C" fn create_transaction(
-    tx_id: c_ulonglong,    // u64
-    offset: *const c_char, // Byte[32] - PrivateKey
-) -> *mut Transaction
+pub unsafe extern "C" fn transactionkernels_add_transaction_kernel(
+    kernels: *mut TransactionKernels,
+    kernel: *mut TransactionKernel,
+) -> bool
 {
+    if kernels.is_null() {
+        return false;
+    }
+
+    if kernel.is_null() {
+        return false;
+    }
+
+    (*kernels).0.push((*kernel).clone());
+    return true;
+}
+
+/// -------------------------------------------------------------------------------------------- ///
+
+/// -------------------------------- Wallet ---------------------------------------------------- ///
+// TODO: Fully implement wallet to finish this off
+
+pub type WalletMasterConfig = WalletConfig;
+
+pub unsafe extern "C" fn create_wallet(
+    // Local Node Identity data
+    config: *const WalletMasterConfig,
+) -> *mut Wallet
+{
+    // TODO do null check for config, runtime
+    let runtime = Runtime::new();
+    let mut w = Wallet::new((*config).clone(), runtime.unwrap());
+    Box::into_raw(Box::new(w.unwrap()))
 }
 
-/// Add a transaction input to a transaction struct
 #[no_mangle]
-pub unsafe extern "C" fn add_transaction_input(
-    transaction: *mut Transaction,
-    commitment: *const c_char, // Byte[32] - Commitment
-    feature_flags: c_char,     // OutputFlags,
-    maturity: c_ulonglong,     // u64
+pub unsafe extern "C" fn start_wallet(wallet: *mut Wallet) -> bool {
+    // (*wallet).start() ? true : false; implement start() on wallet
+    // i.e return (*wallet).start()
+    true
+}
+
+/// Set the Key Manager
+#[no_mangle]
+pub unsafe extern "C" fn wallet_set_key_manager(wallet: *mut Wallet, state: *mut KeyManagerState) -> bool {
+    if wallet.is_null() {
+        return false;
+    }
+
+    if state.is_null() {
+        return false;
+    }
+
+    // (*wallet).key_manager.state = (*state) ? true : false; implement SetState() on Wallet
+    // i.e return (*wallet).SetState((*state));
+    return true;
+}
+
+pub unsafe extern "C" fn wallet_attach_master_seed(wallet: *mut Wallet, seed: *mut KeyManagerSeedWords) -> bool {
```

Comment:
I think `set` is a more appropriate verb for this method.

Reply:
Agreed. Fixed in next commit.